### PR TITLE
Include input element (under HashToGroup) in Finalize DST

### LINKED ISSUE #288: Remove input from Finalize DST
In Finalize, we include `evaluatedElement` in the hash input, which seems redundant with `input`. We could probably drop `input` from this step, which would simplify the API quite a bit too. Or we could replace it with the representative of `input` in the group (`HashToGroup(input)`).

### PR DESCRIPTION
**Pull request by chris-wood**
Closes #288.

### CODE REVIEW

File: draft-irtf-cfrg-voprf.md [hunk 1]
```diff
@@ -1474,8 +1479,8 @@ def DeriveKeyPair(mode, suite, seed):
 ~~~
 seed = a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a
 3a3
-skSm = caeff69352df4905a9121a4997704ca8cee1524a110819eb87deba1a39ec1
-701
+skSm = eeadc25cf9d7b7648aa7aeef6516adc94ab62b6a3fc64dbfbfe386879e691
```
Comment (armfazh) [inline draft-irtf-cfrg-voprf.md:1482]: this secret key is not the same as the one in the JSON file.

**Review by armfazh — APPROVED** [review]
changes are ok; but pls double-check test vectors.

**Comment by bytemare** [comment]
Hi 👋

**Comment by bytemare** [comment]
Hi 👋 
Test-vectors check out with a DST for version 07 👍

**Comment by chris-wood** [comment]
Closing for now. Maybe revisit this later.

**Pull request closed without merging**